File: app/src/main/java/com/lightsoft/parceler/converter/DeviceConverter.java
package com.lightsoft.parceler.converter;

import android.os.Parcel;

import com.lightsoft.parceler.model.Device;

import org.parceler.ParcelConverter;
import org.parceler.Parcels;

public class DeviceConverter implements ParcelConverter<Device> {
    @Override
    public void toParcel(Device input, Parcel parcel) {
        parcel.writeParcelable(Parcels.wrap(input), 0);
    }

    @Override
    public Device fromParcel(Parcel parcel) {
        return Parcels.unwrap(parcel.readParcelable(Device.class.getClassLoader()));
    }
}

File: kotlinparceler/src/main/java/com/lightsoft/kotlinparceler/converter/DeviceConverter.kt
package com.lightsoft.kotlinparceler.converter

import android.os.Parcel
import com.lightsoft.kotlinparceler.model.Device
import org.parceler.ParcelConverter
import org.parceler.Parcels

class DeviceConverter : ParcelConverter<Device> {

    override fun toParcel(input: Device?, parcel: Parcel?) {
        parcel?.writeParcelable(Parcels.wrap(input), 0)
    }

    override fun fromParcel(parcel: Parcel?): Device {
        return Parcels.unwrap(parcel?.readParcelable(Device::class.java.classLoader))
    }
}
